packages/paypal-js/types/v6/components/payments.d.ts

```diff
@@ -0,0 +1,101 @@
+export type ShippingAddressErrorType =
+    | "ADDRESS_ERROR"
+    | "COUNTRY_ERROR"
+    | "STATE_ERROR"
+    | "ZIP_ERROR";
+
+export type ShippingAddressErrorMessages = Record<
+    ShippingAddressErrorType,
+    string
+>;
+
+export type ShippingOptionsErrorType =
+    | "METHOD_UNAVAILABLE"
+    | "STORE_UNAVAILABLE";
+
+export type ShippingOptionsErrorMessages = Record<
+    ShippingOptionsErrorType,
+    string
+>;
+
+export type OnShippingAddressChangeData = {
+    errors: ShippingAddressErrorMessages;
+    orderId: string;
+    shippingAddress: {
+        city: string;
```

Comment:
Technically, countryCode is the only required field and `city`, `postalCode`, and `state` are all optional based on all the different types of addresses around the world.